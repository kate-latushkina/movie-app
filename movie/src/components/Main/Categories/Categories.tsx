import { Tabs, Tab, makeStyles } from "@material-ui/core";
import React, { ChangeEvent, useState } from "react";
import { useDispatch } from "react-redux";
import { setCategory } from "../../../state/reducers/actionCreators";

const useStyles = makeStyles(() => ({
  root: {
    minWidth: 80,
  },
}));

const Categories: React.FC = () => {
  const categoriesArray = ["All", "Action", "Comedy", "Horror", "Crime"];
  const [value, setValue] = useState(0);
  const classes = useStyles();
  const dispatch = useDispatch();

  const handleChange = (_event: ChangeEvent<unknown>, newValue: number) => {
    setValue(newValue);
  };

  const filterCategories = (category: string) => {
    dispatch(setCategory(category));
  };

  return (
    <Tabs
      value={value}
      indicatorColor="primary"
      textColor="primary"
      onChange={handleChange}
      className="categories"
    >
      {categoriesArray.map(category => (
        <Tab
          label={category}
          key={category}
          className={classes.root}
          onClick={() => filterCategories(category)}
        />
      ))}
    </Tabs>
  );
};

export default Categories;
