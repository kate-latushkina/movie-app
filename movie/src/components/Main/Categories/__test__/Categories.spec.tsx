import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import { Tabs } from "@material-ui/core";
import Categories from "../Categories";

describe("should render Categories component", () => {
  let component: any;
  let tabs: any;

  beforeEach(() => {
    component = mount(<Categories />);
    tabs = component.find(Tabs);
  });

  it("should contain Tabs", () => {
    expect(tabs.length).toBe(1);
  });

  it("should render Categories snapshot", () => {
    const wrapper = renderer.create(<Categories />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  describe("should render Tabs state", () => {
    it("Tabs loads with initial state of 0", () => {
      expect(tabs.props().value).toEqual(0);
    });

    it("should handler Tabs set new value", () => {
      tabs.simulate("change");
      tabs.props().value = 1;
      expect(tabs.props().value).toBe(1);
    });
  });
});
