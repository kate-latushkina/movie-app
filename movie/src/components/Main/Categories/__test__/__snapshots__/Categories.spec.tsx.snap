// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Categories component should render Categories snapshot 1`] = `
<div
  className="MuiTabs-root categories"
>
  <div
    className="MuiTabs-scroller MuiTabs-fixed"
    onScroll={[Function]}
    style={
      Object {
        "marginBottom": null,
        "overflow": "hidden",
      }
    }
  >
    <div
      className="MuiTabs-flexContainer"
      onKeyDown={[Function]}
      role="tablist"
    >
      <button
        aria-selected={true}
        className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary makeStyles-root-1 Mui-selected"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="tab"
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiTab-wrapper"
        >
          All
        </span>
        <span
          className="PrivateTabIndicator-root-2 PrivateTabIndicator-colorPrimary-3 MuiTabs-indicator"
          style={Object {}}
        />
      </button>
      <button
        aria-selected={false}
        className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary makeStyles-root-1"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiTab-wrapper"
        >
          Documentary
        </span>
      </button>
      <button
        aria-selected={false}
        className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary makeStyles-root-1"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiTab-wrapper"
        >
          Comedy
        </span>
      </button>
      <button
        aria-selected={false}
        className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary makeStyles-root-1"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiTab-wrapper"
        >
          Horror
        </span>
      </button>
      <button
        aria-selected={false}
        className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary makeStyles-root-1"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiTab-wrapper"
        >
          Crime
        </span>
      </button>
    </div>
  </div>
</div>
`;
